From f06885fab72b1e82dd20dcc29cca67a86a1586a5 Mon Sep 17 00:00:00 2001
From: SuperBFG7 <daniel@despite.ch>
Date: Thu, 22 Oct 2015 11:44:50 +0200
Subject: [PATCH] remember last folder in browser

---
 htdocs/js/mpd.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/htdocs/js/mpd.js b/htdocs/js/mpd.js
index 51e8fdc..a69d97e 100644
--- a/htdocs/js/mpd.js
+++ b/htdocs/js/mpd.js
@@ -329,7 +329,10 @@ function webSocketConnect() {
                         click: function() {
                             switch($(this).attr('class')) {
                                 case 'dir':
-                                    app.setLocation("#/browse/0/"+$(this).attr("uri"));
+                                    pagination = 0;
+                                    browsepath = $(this).attr("uri");
+                                    $("#browse > a").attr("href", '#/browse/'+pagination+'/'+browsepath);
+                                    app.setLocation('#/browse/'+pagination+'/'+browsepath);
                                     break;
                                 case 'song':
                                     socket.send("MPD_API_ADD_TRACK," + decodeURI($(this).attr("uri")));
